### PR TITLE
reformatted result_type #2041

### PR DESCRIPTION
**Pull request by c1boo**

### CODE REVIEW

**Comment by c1boo** [comment]
#2041

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

A few changes are required
1. You have to write docstring example under the container instance method and also container static method.
2. Fix the lint issues caused by your changes.
3. Fix the docstring errors caused by `result_type`.

If you face any trouble or have any question, feel free to ask I'd be happy to help you out.

Thanks.

**Comment by sherry30** [comment]
Hi,

So it seems like there was a problem in the backend, this function is supposed to return ivy.Dtype but was returning the native dtype.
I have pushed a fix for this, now they all should be expecting the same answer i.e.  `{a:float32, b:int32}` which was the one you were receiving for torch before.

File: ivy/container/data_type.py [hunk 1]
```diff
@@ -647,6 +647,45 @@ def static_result_type(
...
+
+        Returns
+        -------
+        ret
+            the dtype resulting from an operation involving the input arrays and dtypes.
+
+        Examples
+        --------
+        >>> x = ivy.Container(a = ivy.array([0, 1, 2]), \
+        b = ivy.array([3., 4., 5.]))
```
Comment (sherry30) [inline ivy/container/data_type.py:680]: indent array `b` exactly under array `a`

File: ivy/container/data_type.py [hunk 2]
```diff
@@ -664,6 +703,46 @@ def result_type(
...
+            the dtype resulting from an operation involving the input arrays and dtypes.
+
+        Examples
+        --------
+        >>> x = ivy.Container(a=ivy.array([0., 1., 2.]), \
+            b=ivy.array([3, 4, 5]))
+        >>> print(x.a.dtype, x.b.dtype)
+        float32 int32
+
+        >>> print(x.result_type(x))
```
Comment (sherry30) [inline ivy/container/data_type.py:740]: running result type on the same containers, maybe create a different container `y` and then run it like `x.result_type(y)`

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

Can you add 1 functional example using container as an input in `ivy/functional/ivy/data_type.py`
I pointed out a few other changes on the particular lines 😃

**Pull request merged**